File: src/main/java/com/itsci/mju/maebanjumpen/repository/PartyRoleRepository.java
package com.itsci.mju.maebanjumpen.repository;

import com.itsci.mju.maebanjumpen.model.PartyRole;
import com.itsci.mju.maebanjumpen.model.Person;
import org.springframework.data.jpa.repository.EntityGraph; // ⬅️ IMPORT EntityGraph
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PartyRoleRepository extends JpaRepository<PartyRole, Integer> {

    @EntityGraph(attributePaths = {"person"})
    List<PartyRole> findByPersonPersonId(Integer personId);
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/partyrole/repository/PartyRoleRepository.kt
package com.itsci.mju.maebanjumpen.partyrole.repository

import com.itsci.mju.maebanjumpen.entity.PartyRole
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PartyRoleRepository : JpaRepository<PartyRole, Long> {

    @EntityGraph(attributePaths = ["person"])
    fun findByPersonId(personId: Long): List<PartyRole>
}

